### DISCUSSION TITLE
How do I add "previous month" and "next month" links to the bottom of the page?

### DISCUSSION THREAD

**Discussion by cellio** [datenstrom/community#700] [Ask a question] [answered]
I am using the PreviousNext extension to add navigation links to individual blog posts, so you don't have to keep going back to the main page if you want to just step through posts one at a time.  How can I add something similar for the month-by-month archive pages?  I can list them all using `[blogarchive]`, and clicking on one of them shows a page that uses the `blogpages` layout but only for that month.  I want to add previous/next links for these month views.

I'm using the pagination layout at the bottom of `blogpages.html`, but that serves a different function: if you have more entries on a page than `BlogPagesMax`, this adds (different) previous/next links.  I suppose I would have a problem if I had more entries in a single month than `BlogPagesMax`, but I don't expect to run into that problem. 

The main blog and the archive month pages (and the individual-tag pages for that matter) all use the `blogpages.html` layout, as far as I can tell.  From within that template, is it possible to tell which kind of "list of blog posts" page I have, so that I could add monthly navigation only to the archive pages?  And how would I get those links -- should I be pattern-matching PreviousNext (for individual entries), or doing something else?  Do I need to modify some PHP?  Make a new template specifically for archive pages and figure out how to hook it in?  Something else?

Thanks for all the help y'all have given me on past questions.  I don't want to be a help vampire and I did try to figure this out myself, but I feel like I'm starting to thrash so I figured it was time to ask for some hints.  (On the plus side, I've gone from "never seen PHP before" to "able to add simple things without burning the whole thing down" in a matter of weeks. :-) )

**Comment by markseuffert** [comment]
I don't think this feature is supported by the [blog extension](https://github.com/annaesvensson/yellow-blog) and the `[blogmonths]` shortcut.

After looking into the code I can see two design problems. The smaller problem is that showing a specific month with the `[blogmonths]` shortcut uses a filter, that effectively removes all other pages. This is the intended behaviour. For example `https://website/blog/published:2022-01/` is a filter for pages published in January 2022. At the time you would like to navigate to a different month, you don't have this information anymore in the HTML layout. The bigger problem is that in order to have a general purpose solution, the navigation between months should work together with the pagination. Otherwise the user interface becomes confusing. Sooner or later. For example when there would be many entries in January 2022, you would see links for `Previous month`, `Previous`, `Next`, `Next month` at the bottom of the page. 

To answer your question. You probably have to make your own shortcut. One where clicking on a specific month jumps to the corresponding position in the middle of the blog. For example January 2022 jumps to `https://website/blog/page:2#my-last-article-in-january`. From there you could use the regular `Previous` and `Next` at the bottom of the page. In theory, this would solve both design problems mentioned above. Just an idea, hope it helps.

**Comment by markseuffert** [comment, marked as answer]
Here's another alternative. You could use the [feed extension](https://github.com/annaesvensson/yellow-feed) as a quick & dirty solution. With a little help of the API you could switch the daily feed output to a monthly feed output. This would visualise the blog as a compact continuous timeline, which is available on your website at `http://website/feed/`. 

In file `system/layouts/feed.html` replace:

```
<?php $sectionNew = $page->getDateHtml("modified") ?>
<?php else: ?>
<?php $sectionNew = $page->getDateHtml("published") ?>
```

....with the following:

```
<?php $sectionNew = $page->getDateHtml("modified", "coreDateFormatShort") ?>
<?php else: ?>
<?php $sectionNew = $page->getDateHtml("published", "coreDateFormatShort") ?>
```

Thanks @schulle4u for the suggestion/brainstorming on Discord. Hope it helps.

**Reply by cellio** [reply]
Thanks to both of you for these suggestions!  I'll give this a shot.
